### PR TITLE
add ability to toggle coinbase transaction scanning

### PR DESCRIPTION
**Pull request by rashedmyt**

### CODE REVIEW

File: src/SettingsScreen.js [hunk 1]
```diff
@@ -230,6 +230,36 @@ export class SettingsScreen extends React.Component {
...
+                        {
+                            title: 'Toggle Coinbase Transactions Scan',
+                            description: 'Enable or disable scanning coinbase transactions. Default is disabled',
+                            icon: {
+                                iconName: 'money-check-alt',
+                                IconType: FontAwesome,
+                            },
+                            onClick: () => {
+                                Alert.alert(
+                                    'Requires Restart',
```
Comment (zpalmtree) [inline src/SettingsScreen.js]: We don't require a restart. If we use `Globals.wallet.scanCoinbaseTransactions(true/false)` it should update without restarting.

File: src/SettingsScreen.js [hunk 2]
```diff
@@ -219,7 +219,7 @@ export class SettingsScreen extends React.Component {
                         },
                         {
                             title: 'Toggle Notifications',
-                            description: 'Enable or disable transaction notifications',
+                            description: 'Enable or disable transaction notifications. Default is enabled',
```
Comment (zpalmtree) [inline src/SettingsScreen.js]: I'd prefer to keep changes focused on the PR, make a separate PR to change this please. (I'm going to alter it anyway)

File: src/CreateScreen.js [hunk 3]
```diff
@@ -82,6 +82,8 @@ export class CreateWalletScreen extends React.Component {
 
     constructor(props) {
         super(props);
+
+        Config.scanCoinbaseTransactions = Globals.preferences.scanCoinbaseTransactions;
```
Comment (zpalmtree) [inline src/CreateScreen.js]: No point setting this here, since the globals aren't loaded from the database until we reach the main screen. Just need to set it when reading from database in Globals.js

File: src/SettingsScreen.js [hunk 4]
```diff
@@ -219,7 +219,7 @@ export class SettingsScreen extends React.Component {
                         },
```
Comment (zpalmtree) [inline src/SettingsScreen.js]: Also, not sure what that red space means, maybe tabs vs spaces? Makes it a tad hard to review changes.

Comment (andrewnk) [reply src/SettingsScreen.js]: I'm on mobile so I can't test, but could it be that the apostrophe need to be escaped?

File: src/SettingsScreen.js (hunk 1 above)
Comment (rashedmyt) [reply src/SettingsScreen.js]: used that method in the latest change

File: src/CreateScreen.js (hunk 3 above)
Comment (rashedmyt) [reply src/CreateScreen.js]: yup.. removed

**Comment by rashedmyt** [comment]
Auto format in VS Code did its magic and now more than required changes are commited _sigh_

**Comment by zpalmtree** [comment]
Maybe I was unclear on the best way to accomplish this. Since preferences is loaded after reaching the create screen, and we can toggle scanning coinbase on/off actively, we just need to add it to the initGlobals() method, and the method in settings.

On loading from initGlobals(), `Globals.wallet.scanCoinbaseTransactions(Globals.preferences.scanCoinbaseTransactions);`, and then in the settings option, something like
```
Globals.preferences.scanCoinbaseTransactions = !Globals.preferences.scanCoinbaseTransactions;
Globals.wallet.scanCoinbaseTransactions(Globals.preferences.scanCoinbaseTransactions);
savePreferencesToDatabase(Globals.preferences);
```

If you see the way the Display Notifications thing is done, it's basically the same as that, just call the method on changing the value as well. We can also do the nice thing where we style the description based on the value...
```
title: Globals.preferences.scanCoinbaseTransactions ? 'Scan coinbase transactions' : 'Skip coinbase transactions'
```
or something like that.
Should be all we need.

**Comment by rashedmyt** [comment]
done..

**Comment by zpalmtree** [comment]
OK, looks good now, I think. Is it easy to revert the whitespace changes?

**Comment by rashedmyt** [comment]
I'll give it a try

**Comment by rashedmyt** [comment]
removed whitespace changes but those red lines aren't going

**Comment by zpalmtree** [comment]
Yeah, I think that's because we're using jsx instead of js, which it doesn't like. Might go away if we rename the file extension to .jsx, I'll look into. Looking good now. Can we maybe change the description to something like 'Enable this if you have ever solo mined a block', or something similar.

I think the average user won't know what coinbase transactions are, and we don't want them shooting themselves in the foot.

**Comment by rashedmyt** [comment]
done.. I seriously need to be creative regarding writing documentation and descriptions

**Comment by zpalmtree** [comment]
Looks good. Thanks for the PR!
